pkg/lock/memory.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2023 The KubeService-Stack Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lock
+
+import (
+	"runtime"
+	"sync/atomic"
+)
+
+// MemoryLock implements sync/Locker, default 0 indicates an unlocked memory.
+type MemoryLock struct {
+	_flag uint32
+}
+
```

Comment:
add `NewMemoryLock` func

Reply:
THKS! Done. Add `NewMemoryLock` function

Reply:
Thinks for you review! Done. Add `NewMemoryLock` create function